### ISSUE TITLE
rrulestr ignore DTSTART following elements

### ISSUE THREAD

**Issue by Symonen** [dateutil/dateutil#942]
`
In[33]: from dateutil import rrule; import datetime;
In[34]: now = datetime.datetime.now()
In[35]: r1, r2 = rrule.rrule(rrule.DAILY, count=10, dtstart=now), rrule.rrule(rrule.DAILY, count=10, dtstart=now + datetime.timedelta(5))
In[36]: rs = rrule.rruleset()
In[37]: rs.rrule(r1)
In[38]: rs.rrule(r2)
In[39]: rs_str = '{}\n{}'.format(r1, r2)
In[40]: rs_str
Out[40]: 'DTSTART:20190802T093457\nRRULE:FREQ=DAILY;COUNT=10\nDTSTART:20190807T093457\nRRULE:FREQ=DAILY;COUNT=10'
In[41]: rs2 = rrule.rrulestr(rs_str)
In[42]: set(rs.between(now,  now + datetime.timedelta(10))) - set(rs2.between(now,  now + datetime.timedelta(10)))
Out[42]: 
{datetime.datetime(2019, 8, 3, 9, 34, 57),
 datetime.datetime(2019, 8, 4, 9, 34, 57),
 datetime.datetime(2019, 8, 5, 9, 34, 57),
 datetime.datetime(2019, 8, 6, 9, 34, 57)}
In[43]: rs.between(now,  now + datetime.timedelta(10))
Out[43]: 
[datetime.datetime(2019, 8, 3, 9, 34, 57),
 datetime.datetime(2019, 8, 4, 9, 34, 57),
 datetime.datetime(2019, 8, 5, 9, 34, 57),
 datetime.datetime(2019, 8, 6, 9, 34, 57),
 datetime.datetime(2019, 8, 7, 9, 34, 57),
 datetime.datetime(2019, 8, 8, 9, 34, 57),
 datetime.datetime(2019, 8, 9, 9, 34, 57),
 datetime.datetime(2019, 8, 10, 9, 34, 57),
 datetime.datetime(2019, 8, 11, 9, 34, 57),
 datetime.datetime(2019, 8, 12, 9, 34, 57)]
In[44]: rs2.between(now,  now + datetime.timedelta(10))
Out[44]: 
[datetime.datetime(2019, 8, 7, 9, 34, 57),
 datetime.datetime(2019, 8, 8, 9, 34, 57),
 datetime.datetime(2019, 8, 9, 9, 34, 57),
 datetime.datetime(2019, 8, 10, 9, 34, 57),
 datetime.datetime(2019, 8, 11, 9, 34, 57),
 datetime.datetime(2019, 8, 12, 9, 34, 57)]

In[45]: dateutil.__version__
Out[47]: '2.8.0'
In[48]: sys.version
Out[49]: '2.7.16 (default, Jul 13 2019, 16:01:51) \n[GCC 8.3.0]'
`
The `dateutil.rrule.rrulestr` ignores second DTSTART and applies the first one to all `rrules` in `rruleset`. It should respect all DTSTART and set them properly to every `rrule`?

**Comment by pganssle** [comment]
@Symonen Thank you for your report, sorry for the delay in responding.

I will have to look into this. I am mildly surprised that you can construct an `rruleset` by concatentating `RRULE`s like this; I think the best thing to do is to check how RRULE sets are normally represented (if at all) in [RFC 5545](https://tools.ietf.org/html/rfc5545), and hopefully that will give us a clue as to whether we should support `DTSTART` or change the way `rruleset` can be parsed from a string.

**Comment by ffe4** [comment]
I don't think we can rely on the RFC for this. From [3.8.5.3.  Recurrence Rule](https://tools.ietf.org/html/rfc5545#section-3.8.5.3):
>  This property can be specified in recurring [calendar components and sub-components], but it SHOULD NOT be specified more than once. The recurrence set generated with multiple "RRULE" properties is undefined.

Since DTSTART is not part of the RRULE in the RFC, but of the surrounding component, there should probably be only one DTSTART in a `ruleset`.
